app/models/Property.ts

```diff
@@ -81,18 +87,62 @@ export class Property {
   private calculateNewBuildPrice() {
     let newBuildPrice = this.newBuildPricePerMetre * this.size;
     newBuildPrice = parseFloat(newBuildPrice.toFixed(PRECISION));
-
     return newBuildPrice;
   }
 
   private calculateDepreciatedBuildPrice() {
-    let depreciatedBuildPrice =
-      this.newBuildPrice + DEPRECIATION_FACTOR * math.log(this.age);
-    depreciatedBuildPrice = parseFloat(
-      depreciatedBuildPrice.toFixed(PRECISION)
+    if (this.age === 0) return this.newBuildPrice; // If newbuild, return newBuildPrice and don't depreciate
+    
+    let depreciatedBuildPrice = 0;
+
+  // Calculate for each component using the public method
+  for (const key of Object.keys(HOUSE_BREAKDOWN_PERCENTAGES) as (keyof houseBreakdownType)[]) {
+    const result = this.calculateComponentValue(
+      key, 
+      this.newBuildPrice, 
+      this.age, 
+      MAINTENANCE_LEVELS[0]
     );
 
-    return depreciatedBuildPrice;
+    depreciatedBuildPrice += result.depreciatedComponentValue;
+  }
+    depreciatedBuildPrice = parseFloat(depreciatedBuildPrice.toFixed(PRECISION))
+
+  return depreciatedBuildPrice;
+}
+
+  public calculateComponentValue(
+    componentKey: keyof houseBreakdownType,
+    newBuildPrice: number,
+    age: number,
+    maintenanceLevel: number
+  ): ComponentCalculation {
+    const component = HOUSE_BREAKDOWN_PERCENTAGES[componentKey];
+    
+    // Calculate new component value
+    const newComponentValue = newBuildPrice * component.percentageOfHouse;
+    
+    // Calculate depreciation
+    const depreciationFactor = 1 - (component.depreciationPercentageYearly * age);
+    
+    // Calculate maintenance (0 for foundations and structure)
+    const maintenanceAddition = 
+      (componentKey === 'foundations' || componentKey === 'structureEnvelope') 
+        ? 0 
+        : maintenanceLevel * newBuildPrice * age * component.percentOfMaintenanceYearly;
+    
+    // Calculate final value
+    let depreciatedComponentValue = 
+      (newComponentValue * depreciationFactor) + maintenanceAddition;
+
+    depreciatedComponentValue < 0 ? depreciatedComponentValue = 0 : depreciatedComponentValue
```

Comment:
```suggestion
    // Do not depreciate below 0
    if (depreciatedComponentValue < 0) depreciatedComponentValue = 0
```

Maybe a bit clearer?

Reply:
Much clearer 😅 thanks!